Add OMIE integration (#40448)
home-assistant/home-assistant.github.io: source/_integrations/omie.markdown

---
title: OMIE

> description: Instructions on how to integrate OMIE day-ahead market prices within Home Assistant.

Reviewer: ```suggestion
description: Monitor OMIE day-ahead electricity market prices for Spain and Portugal directly in Home Assistant.
```

ha_category:
  - Energy
  - Sensor
ha_release: '2026.5'
ha_iot_class: Cloud Polling
ha_quality_scale: bronze
ha_config_flow: true
ha_codeowners:
  - '@luuuis'
ha_domain: omie
ha_platforms:
  - sensor
ha_integration_type: service
---

[OMIE](https://www.omie.es/en) is the electricity market operator for the day-ahead and intraday energy markets in Spain and Portugal. The **OMIE** {% term integration %} retrieves day-ahead market prices from the OMIE APIs and makes them available within Home Assistant.

Having the wholesale electricity prices within Home Assistant enables a range of use cases, such as:

- Calculating electricity bills ahead of time (for those on variable-price tariffs that are linked to the wholesale price)
- Deciding whether to export locally produced energy (for example from solar panels) to the grid

{% include integrations/config_flow.md %}

## Sensors

The **OMIE** integration retrieves the results of the [day-ahead market](https://www.omie.es/en/mercado-de-electricidad) for Spain and Portugal daily and exposes them as the following {% term sensors %}.

> - Portugal spot price

Reviewer: ```suggestion
- **Portugal spot price**: Current quarter-hour electricity spot price in Portugal, in €/kWh.
```
Maybe add a description here? Please check if the suggestion is correct.

> - Spain spot price

Reviewer: ```suggestion
- **Spain spot price**: Current quarter-hour electricity spot price in Spain, in €/kWh.
```

> 

Reviewer: ## Data updates

The OMIE integration {% term polling polls %} OMIE for new data at the start of each quarter-hour (in Central European Time).

---
title: OMIE - Spain and Portugal electricity market data
description: Instructions on how to integrate OMIE day-ahead market prices within Home Assistant.
ha_category:
  - Energy
  - Sensor
ha_release: '2025.10'
ha_iot_class: Cloud Polling
ha_quality_scale: Bronze
ha_config_flow: true
ha_codeowners:
  - '@luuuis'
ha_domain: omie
ha_platforms:
  - sensor

> ha_integration_type: integration

Reviewer: ```suggestion
ha_integration_type: service
```

---

The **OMIE** {% term integration %} retrieves day-ahead market results from [OMIE](https://www.omie.es/en) APIs
 and makes them available within Home Assistant. [OMIE](https://www.omie.es/en) is the electricity market operator for Spain and Portugal's day-ahead and intraday energy markets. Having the wholesale electricity prices within Home Assistant enables a range of use cases, such as:

- calculating electricity bills ahead of time (for those on variable-price tariffs that are linked
  to the wholesale price),

- deciding whether to export locally-produced energy (for example from solar panels) to the grid
  depending on remuneration.

> 

Reviewer: Are there any prerequisites? Any thing I need to have or do before adding the integration? An account, and app? Do I need to enter any information when I add the integration, like login credentials?

Author: There are no prerequisites, this is published (and public) energy market data.

https://www.omie.es/en/market-results/daily/daily-market/day-ahead-price

No account needed.

> {% include integrations/config_flow.md %}

Reviewer: this textblock is rendered like this:
<img width="695" height="240" alt="image" src="https://github.com/user-attachments/assets/faaffe48-6680-477b-969f-57fd604340a5" />

## Sensors

The **OMIE** integration will retrieve the results of the [day-ahead market](https://www.omie.es/en/mercado-de-electricidad) for Spain and Portugal on a daily basis and expose them to the following {% term sensors %}.

- **Portugal spot price**: the wholesale price in the day-ahead market for the current hour in Portugal (in €/KWh)
- **Spain spot price**: the wholesale price in the day-ahead market for the current hour in Spain (in €/KWh)

> 

Reviewer: I know you mentioned here the daily basis, but would it be worth adding a section on [data updates](https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown#data-updates)?

---
title: OMIE - Spain and Portugal electricity market data
description: Instructions on how to integrate OMIE day-ahead market prices within Home Assistant.
ha_category:
  - Energy
  - Sensor
ha_release: '2025.9'
ha_iot_class: Cloud Polling
ha_quality_scale: Silver
ha_config_flow: true
ha_codeowners:
  - '@luuuis'
ha_domain: omie
ha_platforms:
  - sensor
ha_integration_type: integration
---

The **OMIE** {% term integration %} retrieves day-ahead market results from [OMIE](https://www.omie.es/en) APIs
 and makes them available within Home Assistant. [OMIE](https://www.omie.es/en) is the electricity market operator for Spain and Portugal's day-ahead and intraday energy markets. Having the wholesale electricity prices within Home Assistant enables a range of use cases, such as:

- calculating electricity bills ahead of time (for those on variable-price tariffs that are linked
  to the wholesale price),

- deciding whether to export locally-produced energy (for example from solar panels) to the grid
  depending on remuneration.

> 

Reviewer: ```suggestion
```

{% include integrations/config_flow.md %}

## Sensors

The **OMIE** integration will retrieve the results of the [day-ahead market](https://www.omie.es/en/mercado-de-electricidad) for Spain and Portugal on a daily basis and expose them to the following {% term sensors %}.

- **Marginal Price - Portugal**: the wholesale price in the day-ahead market for the current hour Portugal (€/KWh)

> - **Marginal Price - Spain**: the wholesale price in the day-ahead market for the current hour Spain (€/KWh)

Reviewer: ```suggestion
- **Marginal price: Portugal**: the wholesale price in the day-ahead market for the current hour for Portugal (in €/kWh)
- **Marginal price: Spain**: the wholesale price in the day-ahead market for the current hour for Spain (in €/kWh)
```
We usually use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization), also in UI strings.

- on another note: [in the code](https://github.com/home-assistant/core/pull/150399/files#diff-e2e32aa1cb039d119621084b5f3cdba229451df1829f4a4abcc2548b190c6a47R12-R18), they seem to be called "spot price", not marginal price. Are these the same strings?

Author: Indeed, I forgot to come back here and update the sensor names. Thank you.

Edit: updated the sensor name in HA/omie strings to match.

---
title: OMIE
description: Monitor OMIE day-ahead electricity market prices for Spain and Portugal directly in Home Assistant.
ha_category:
  - Energy
  - Sensor

> ha_release: '2026.5'

Reviewer: ```suggestion
ha_release: 2026.5
```

---
title: OMIE - Spain and Portugal electricity market data
description: Instructions on how to integrate OMIE day-ahead market prices within Home Assistant.
ha_category:
  - Energy
  - Sensor
ha_release: '2025.9'
ha_iot_class: Cloud Polling
ha_quality_scale: Silver
ha_config_flow: true
ha_codeowners:
  - '@luuuis'
ha_domain: omie
ha_platforms:
  - sensor
ha_integration_type: integration
---

> [OMIE](https://www.omie.es/en) is the electricity market operator for Spain and Portugal's day-ahead and
> intraday energy markets.
> 

Reviewer: ```suggestion
```

> The **OMIE** {% term integration %} retrieves day-ahead market results from [OMIE](https://www.omie.es/en) APIs
>  and makes them available within Home Assistant. Having the wholesale electricity prices within Home Assistant
>  in turn enables a range of use cases, such as:

Reviewer: ```suggestion
The **OMIE** {% term integration %} retrieves day-ahead market results from [OMIE](https://www.omie.es/en) APIs
 and makes them available within Home Assistant. [OMIE](https://www.omie.es/en) is the electricity market operator for Spain and Portugal's day-ahead and intraday energy markets. Having the wholesale electricity prices within Home Assistant enables a range of use cases, such as:
```

- calculating electricity bills ahead of time (for those on variable-price tariffs that are linked
  to the wholesale price),

- deciding whether to export locally-produced energy (for example from solar panels) to the grid
  depending on remuneration.

> ## Configuration
> 
> To add the **OMIE** integration to your Home Assistant instance, use this My button:
> 
> [![Open your Home Assistant instance and start setting up the OMIE integration.](https://my.home-assistant.io/badges/config_flow_start.svg)](https://my.home-assistant.io/redirect/config_flow_start/?domain=omie)

Reviewer: ```suggestion
{% include integrations/config_flow.md %}
```
We have a predefined text block for this.
